src/include/miopen/fold/problem_description.hpp

```diff
@@ -0,0 +1,481 @@
+                         dilation[i] * (kernel_size[i] - 1) - 1) /
+                            stride[i] +
+                        1;
+            L *= l;
+            ls.push_back(l);
+        }
+        std::vector<size_t> output_dims_desired{
+            static_cast<size_t>(N), static_cast<size_t>(C * P), static_cast<size_t>(L)};
+        auto output_dims = outputDesc.GetLengths();
+        if(output_dims != output_dims_desired)
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Unfold: Invalid output dimension");
+#else
+            return false;
+#endif
+        }
+        return true;
+    }
+
+    bool IsValidType() const
+    {
+        if(inputDesc.GetType() != outputDesc.GetType())
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "Unfold: The input tensor and output tensor has mismatch type.");
+#else
+            return false;
+#endif
+        }
+        return true;
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+public:
+    TensorDescriptor inputDesc;
+    TensorDescriptor outputDesc;
+    const int32_t* kernel_size;
+    const int kernel_size_size;
+    const int32_t* stride;
+    const int stride_size;
+    const int32_t* padding;
+    const int padding_size;
+    const int32_t* dilation;
+    const int dilation_size;
+};
+
+struct UnfoldBwdProblemDescription : ProblemDescriptionBase
+{
+    UnfoldBwdProblemDescription(const TensorDescriptor& dinputDesc_,
+                                const TensorDescriptor& doutputDesc_,
+                                const int32_t* kernel_size_,
+                                const int kernel_size_size_,
+                                const int32_t* stride_,
+                                const int stride_size_,
+                                const int32_t* padding_,
+                                const int padding_size_,
+                                const int32_t* dilation_,
+                                const int dilation_size_)
+        : dinputDesc(dinputDesc_),
+          doutputDesc(doutputDesc_),
+          kernel_size(kernel_size_),
+          kernel_size_size(kernel_size_size_),
+          stride(stride_),
+          stride_size(stride_size_),
+          padding(padding_),
+          padding_size(padding_size_),
+          dilation(dilation_),
+          dilation_size(dilation_size_)
+    {
+        IsValidSize();
+        IsValidType();
+    }
+
+    bool IsValidSize() const
+    {
+        if(dinputDesc.GetSize() != 4)
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Unfold: The input gradient tensor should be 4D.");
+#else
+            return false;
```

Comment:
Same

Reply:
resolve in commit [8ee2861](https://github.com/moreh-dev/MIOpen/pull/37/commits/8ee286160f08a51be318531311032182c8cb3a12)